### DISCUSSION TITLE
GC问题。

### DISCUSSION THREAD

**Discussion by lsx1994** [cloudwu/skynet#1540] [Q&A] [answered]
请问skynet是在什么时候执行collectgarbage呢？
我开了10000个服务，每秒执行：local a = {}
内存不断的往上涨，看了大概有十几分钟。

具体点说，collectgarbage("collect") 需要我自己调用吗？
或者说skynet只有在退出服务的时候才会调用collectgarbage("collect")吗？

**Comment by cloudwu** [comment, marked as answer]
skynet 秉承标准 Lua 的逻辑：Lua 的 GC 是分配驱动的，只有申请新内存才推进 GC 。
skynet 每个服务都是独立的 Lua VM , 每个 VM 的 GC 是相互独立的，以标准 Lua 的形式工作。

**Comment by lsx1994** [comment]
明白了， 感谢大佬回复

**Comment by sniper00** [comment]
我们是 长期存在的服务用 分代模式，agent用增量式。线上测试内存比较稳定，不会无限增长

**Comment by Gowa2017** [comment]
学到了

Bruce ***@***.***> 于 2022年2月21日周一 18:14写道：
